src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
```

Comment:
I'm on board with avoiding exceptions because the specific assembly isn't available on a given endpoint. But according to the linked issue, this would also silently ignore spelling errors although a given type might be available. The later scenario doesn't seem to be something we would want to swallow in my opinion, but at that point, I'm not sure how we could limit ourselves to the first scenario?